test/mocks/plugins/DirectCallPlugin.sol

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.19;
+
+import {PluginManifest, PluginMetadata} from "../../../src/interfaces/IPlugin.sol";
+import {IStandardExecutor} from "../../../src/interfaces/IStandardExecutor.sol";
+import {IExecutionHook} from "../../../src/interfaces/IExecutionHook.sol";
+
+import {BasePlugin} from "../../../src/plugins/BasePlugin.sol";
+
+contract DirectCallPlugin is BasePlugin, IExecutionHook {
+    bool public preHookRan = false;
+    bool public postHookRan = false;
+
+    function onInstall(bytes calldata) external override {}
+
+    function onUninstall(bytes calldata) external override {}
+
+    function pluginManifest() external pure override returns (PluginManifest memory) {}
+
+    function directCall() external returns (bytes memory) {
+        return IStandardExecutor(msg.sender).execute(address(this), 0, abi.encodeCall(this.getData, ()));
+    }
+
+    function getData() external pure returns (bytes memory) {
+        return hex"04546b";
+    }
+
+    function pluginMetadata() external pure override returns (PluginMetadata memory) {}
+
+    function preExecutionHook(uint8, address sender, uint256, bytes calldata)
+        external
+        override
+        returns (bytes memory)
+    {
+        require(sender == address(this), "mock direct call pre permission hook failed");
```

Comment:
Ugh this is an annoying linter warning because it's in a test. Maybe add an exclusion for the rule `reason-string` in tests?

Reply:
Addressed in [0f04d85](https://github.com/erc6900/reference-implementation/pull/90/commits/0f04d85bdfbd1a52832d681055a760093a678cb5)!

Doubled line length